src/parser/SparqlParser.cpp

```diff
@@ -168,47 +168,54 @@ void SparqlParser::parseSelect(ParsedQuery* query) {
     selectClause._reduced = true;
   }
   if (_lexer.accept("*")) {
-    selectClause._varsOrAsterisk.setsAsterisk();
+    selectClause._varsOrAsterisk.setAllVariablesSelected();
   }
+  std::vector<std::string> manuallySelectedVariables;
   while (!_lexer.accept("where")) {
-    if (_lexer.accept(SparqlToken::Type::VARIABLE)) {
-      // Exception avoided due to previous Syntax Check of Selector '*'
-      selectClause._varsOrAsterisk.getSelectVariables().push_back(
-          _lexer.current().raw);
-    } else if (_lexer.accept("text")) {
-      _lexer.expect("(");
-      std::ostringstream s;
-      s << "TEXT(";
-      _lexer.expect(SparqlToken::Type::VARIABLE);
-      s << _lexer.current().raw;
-      _lexer.expect(")");
-      s << ")";
-      selectClause._varsOrAsterisk.getSelectVariables().push_back(s.str());
-    } else if (_lexer.accept("score")) {
-      _lexer.expect("(");
-      std::ostringstream s;
-      s << "SCORE(";
-      _lexer.expect(SparqlToken::Type::VARIABLE);
-      s << _lexer.current().raw;
-      _lexer.expect(")");
-      s << ")";
-      selectClause._varsOrAsterisk.getSelectVariables().push_back(s.str());
-    } else if (_lexer.accept("(")) {
-      // expect an alias
-      ParsedQuery::Alias a = parseAliasWithAntlr();
-      selectClause._aliases.push_back(a);
-      selectClause._varsOrAsterisk.getSelectVariables().emplace_back(
-          a._outVarName);
-      _lexer.expect(")");
+    if (selectClause._varsOrAsterisk.isAsterisk()) {
+      if (!_lexer.empty()) {
+        throw ParseException("Keyword WHERE expected after SELECT '*' ");
+      }
     } else {
```

Comment:
With my change from above, you don't need the `else` here, because the `if` above always throws an exception.